src/Core/Layer/LayerUpdateStrategy.js

```diff
@@ -46,17 +46,29 @@ function _dichotomy(nodeLevel, currentLevel, options) {
         Math.ceil((currentLevel + nodeLevel) / 2));
 }
 
-export function chooseNextLevelToFetch(strategy, node, nodeLevel, currentLevel, layer) {
+export function chooseNextLevelToFetch(strategy, node, nodeLevel, currentLevel, layer, failureParams) {
+    let nextLevelToFetch;
+    const maxZoom = layer.options.zoom ? layer.options.zoom.max : Infinity;
     switch (strategy) {
         case STRATEGY_GROUP:
-            return _group(nodeLevel, currentLevel, layer.updateStrategy.options);
-        case STRATEGY_PROGRESSIVE:
-            return _progressive(nodeLevel, currentLevel, layer.updateStrategy.options);
+            nodeLevel = failureParams ? Math.ceil((currentLevel + failureParams.targetLevel) / 2) : nodeLevel;
```

Comment:
Why do you override `nodeLevel` in different ways?
```js
nodeLevel = failureParams ? Math.ceil((currentLevel + failureParams.targetLevel) / 2) : nodeLevel;
// vs
nodeLevel = failureParams ? failureParams.targetLevel : nodeLevel;
```
I fail to see a good reason to not always use the same formula.

Reply:
by example 
* for STRATEGY_DICHOTOMY : there'll be a double dichotomies
* for STRATEGY_GROUP: it would return `failureParams.targetLevel`

Reply:
* for STRATEGY_DICHOTOMY: 
so for this strategy with `nodeLevel = failureParams ? failureParams.targetLevel : nodeLevel;` you to get your result `7`

* for STRATEGY_GROUP: we use `nodeLevel = failureParams ? Math.ceil((currentLevel + failureParams.targetLevel) / 2) : nodeLevel` to get next level to try and then we use it in `_group` to fetch one of the values from the groups defined by the user.